Source/WebKit/WebProcess/cocoa/VideoPresentationManager.mm

```diff
@@ -192,36 +192,37 @@ static FloatRect inlineVideoFrame(HTMLVideoElement& element)
     return std::make_tuple(WTFMove(model), WTFMove(interface));
 }
 
-VideoPresentationManager::ModelInterfaceTuple& VideoPresentationManager::ensureModelAndInterface(PlaybackSessionContextIdentifier contextId, bool createlayerHostingContext)
+const VideoPresentationManager::ModelInterfaceTuple& VideoPresentationManager::ensureModelAndInterface(PlaybackSessionContextIdentifier contextId, bool createlayerHostingContext)
 {
-    auto addResult = m_contextMap.add(contextId, ModelInterfaceTuple());
-    if (addResult.isNewEntry)
-        addResult.iterator->value = createModelAndInterface(contextId, createlayerHostingContext);
+    auto it = m_contextMap.find(contextId);
+    if (it != m_contextMap.end())
+        return it->value;
+    auto addResult = m_contextMap.add(contextId, createModelAndInterface(contextId, createlayerHostingContext));
+    ASSERT(addResult.isNewEntry);
     return addResult.iterator->value;
 }
 
-WebCore::VideoPresentationModelVideoElement& VideoPresentationManager::ensureModel(PlaybackSessionContextIdentifier contextId)
+Ref<WebCore::VideoPresentationModelVideoElement> VideoPresentationManager::ensureModel(PlaybackSessionContextIdentifier contextId)
 {
-    return *std::get<0>(ensureModelAndInterface(contextId));
+    return std::get<0>(ensureModelAndInterface(contextId));
 }
 
-VideoPresentationInterfaceContext& VideoPresentationManager::ensureInterface(PlaybackSessionContextIdentifier contextId)
+Ref<VideoPresentationInterfaceContext> VideoPresentationManager::ensureInterface(PlaybackSessionContextIdentifier contextId)
 {
-    return *std::get<1>(ensureModelAndInterface(contextId));
+    return std::get<1>(ensureModelAndInterface(contextId));
 }
 
 void VideoPresentationManager::removeContext(PlaybackSessionContextIdentifier contextId)
 {
     m_playbackSessionManager->removeClientForContext(contextId);
 
-    auto [model, interface] = m_contextMap.get(contextId);
-    ASSERT(model);
-    ASSERT(interface);
-    if (!model || !interface)
+    auto it = m_contextMap.find(contextId);
```

Comment:
Shouldn't we assert that m_contextMap contains contextId?

Reply:
the code before only asserted that model and interface weren't null, and they can't ever be now.

Reply:
on further thought, I re-added the asserts